generate-meson.py

```diff
@@ -0,0 +1,198 @@
+# Small script that generates a meson.build file in the given folder.
```

Comment:
I think this should become a module of `sage_setup`: Like `sage_setup.autogen`, in generates source code.

Reply:
This script is no longer needed once the initial set of meson files is generated. I'll remove it once the PR has been (or is close to be) positively reviewed.

Reply:
I disagree. 

This would put the unnecessary burden on developers to maintain the `meson.build` files.

Reply:
The auto-generated files are missing a lot of important stuff. it was convenient to have them generate the initial boilerplate, but in the future, we'll be changing the build files one line at a time.

Reply:
What is missing that cannot be autogenerated?